tls/s2n_server_hello.c

```diff
@@ -187,7 +187,8 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
         return 0;
     }
 
-    if (IS_RESUMPTION_HANDSHAKE(conn->handshake.handshake_type)) {
+    if (IS_RESUMPTION_HANDSHAKE(conn->handshake.handshake_type)
+        && s2n_connection_get_protocol_version(conn) < S2N_TLS13) {
```

Comment:
Are you sure this fixes the problem? We make the same check for the same reason in [s2n_server_finished](https://github.com/aws/s2n-tls/blob/c72576f6fdc676d0b747ae9add033829c1d3e5c8/tls/s2n_server_finished.c#L70).

There's also a non-macro version of this check in [s2n_resume.c](https://github.com/aws/s2n-tls/blob/c72576f6fdc676d0b747ae9add033829c1d3e5c8/tls/s2n_resume.c#L344-L348) which looks like it affects [whether or not we send the "server_name" extension](https://github.com/aws/s2n-tls/blob/c72576f6fdc676d0b747ae9add033829c1d3e5c8/tls/extensions/s2n_server_server_name.c#L34-L37)-- is that the right behavior in TLS1.3?

Reply:
I would rename the macro to IS_TLS12_RESUMPTION_HANDSHAKE and add the protocol version check inside of that macro to make sure we can't mis-use the macro.

Reply:
This would fix the non-macro function for now, as that actually calls the macro. We are going to have to revamp the non-macro function to work for TLS1.3, but since we don't need it in the external psk code I think that can wait.

Reply:
The macro is really just supposed to operate on the handshake type, not the connection. If we're going to do that, we should probably fix it in the non-macro function and use that instead of the macro. But we'd need to verify that everywhere that checks for resumption with either method really is just checking for TLS1.2 resumption, and that the checks shouldn't also apply to TLS1.3 PSKs.